samcli/commands/remote/remote_invoke_context.py

```diff
@@ -242,7 +242,7 @@ class DefaultRemoteInvokeResponseConsumer(RemoteInvokeConsumer[RemoteInvokeRespo
     _stream_writer: StreamWriter
 
     def consume(self, remote_invoke_response: RemoteInvokeResponse) -> None:
-        self._stream_writer.write(cast(str, remote_invoke_response.response).encode())
+        self._stream_writer.write_bytes(cast(str, remote_invoke_response.response).encode())
```

Comment:
nit: could this be a method in `StreamWriter` where it encodes the given string and writes as bytes (just a QOL improvement so that we don't need to encode every time calling it)

edit: I saw that there is also `write_str` method below, why we are not using it here? (trying to understand the reasoning)

Reply:
This mostly comes down to time and understanding. I didn't want to do a complete refactor within this commit but instead add typing to help myself get this right and solve the bug attached to this commit. 

I could have gone through and update all of this but it's not clear to me why we are writing both `str` and `bytes` to the stream. This might be legacy at this point but I wanted to keep the paths the same as much as I could for now. Accepting `bytes` likely stems from docker returning logs as `bytes`. I vaguely recall my past struggles with this area during the Py2 to Py3 migration but that was so long ago now, those struggles were replaced 😊. My preference here is to do this migration to only `str` slowly to hopefully not affect the content we write to the console.